### ISSUE TITLE
Different INPUT behavior comparing to QBasic or QB64

### ISSUE THREAD

**Issue by afbytes** [boxgaming/qbjs#34]
The results of running the following piece of code on QBasic and QBJS are different.

```bas
INPUT "Enter a number: ", x
 
x = x + 5
PRINT "X is now: "; x
```

Regards to QBasic,

* it considers `x` as a **number** instead of a **string**;
  * note: if you use a variable name `x$`, then it will accept a string.
* if you input "abc" and press ENTER, it will prompt you with `Redo from start` and ask you to input again;
  * note: empty strings are allowed.

Regards to QBJS,

* it considers variable `x` as a **string**;
* when user input "abc", the PRINT result will be `abc5`.

This issue may be related to #33.

**Comment by boxgaming** [comment]
Thanks for reporting the issue @afbytes!  I've added it to the list for the next release.  You can see the list of what is being worked on next on the [Roadmap](https://github.com/boxgaming/qbjs/wiki/Roadmap) page in the wiki.

**Comment by boxgaming** [comment]
Hi @afbytes, this issue has been resolved as of 0.6.0.  The input method will now convert the entered value to the appropriate type.  At present it is still required to declare your variables for this to work properly.  Support for implicit variable declaration is planned for a future release.  The following will now work as expected:
```basic
DIM x
INPUT "Enter a number: ", x
 
x = x + 5
PRINT "X is now: "; x
```
